### ISSUE TITLE
WCAG compliance

### ISSUE THREAD

**Issue by guilheb** [primefaces/primevue#7571]
### Describe the bug

According to the Chrome plugin [Accessible Web Helper](https://chromewebstore.google.com/detail/accessible-web-helper/gdnpkbipbholkoaggmlblpbmgemddbgb?hl=en&pli=1), there are several components that have issues classified as "critical" or "serious".

For this bug report, I have visited the documentation pages on primevue.org as of today and scanned the pages using Accessible Web Helper. I have highlighted the exact issues reported by the plugin.

![Image](https://github.com/user-attachments/assets/6f4f2c9b-4052-45e9-941d-b092c7c5a5d7)
![Image](https://github.com/user-attachments/assets/2b9b6dff-32af-4fc5-aa80-e2182bf6ce23)
![Image](https://github.com/user-attachments/assets/3109d2fc-8bf1-45af-9301-b008971bf9bd)
![Image](https://github.com/user-attachments/assets/948d75cf-67d9-4c76-9082-08978ab9827a)
![Image](https://github.com/user-attachments/assets/552df506-e151-4392-9dd1-f1f5fe485230)
![Image](https://github.com/user-attachments/assets/90ce08af-d63d-430c-b965-18cec6d19540)

### Reason for not contributing a PR

- [ ] Lack of time
- [x] Unsure how to implement the fix/feature
- [x] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://stackblitz.com/edit/primevue-4-vite-issue-template-6tvshcbc?file=src%2FApp.vue

### Environment

Issues were found on the PrimeVue documentation as of today.

### Vue version

N/A

### PrimeVue version

4.3.3

### Node version

N/A

### Steps to reproduce the behavior

1. install the [Accessible Web Helper](https://chromewebstore.google.com/detail/accessible-web-helper/gdnpkbipbholkoaggmlblpbmgemddbgb?hl=en&pli=1) Chrome plugin
2. go to the PrimeVue documentation page
3. navigate to Password, ProgressBar, Select, Stepper, Tree or TreeTable components
4. scan the page using the Accessible Web Helper plugin

### Expected behavior

No critical or serious issues regarding the components themselves.

**Comment by cagataycivici** [comment]
If you could provide PRs, we'd be happy to review and merge.

**Comment by cagataycivici** [comment]
Please create separate tickets for each enhancement instead of a single issue as each case needs to be handled differently.
